templates/flavors/aks/cluster-template.yaml

```diff
@@ -109,3 +109,17 @@ spec:
   enableNodePublicIP: false
   scaleSetPriority: Regular
   name: pool1
+  kubeletConfig:
```

Comment:
Good question:

1) prefer a value that *isn't* the AKS default
2) so long as that value isn't too interesting (and could thus interfere with other features and/or E2E tests)

Basically: wanting to make sure that we engage the actual feature in some way (for AKS to actually build a "configured" node pool) without providing a materially different operational outcome.

Are we concerned that the non-tested "reference" template is getting too interesting? Or do we prefer to expose as much configuration as possible as a feature reference? We can always simplify these configurations for the reference templates and triage them to the `test/ci` templates only...

Reply:
> Are we concerned that the non-tested "reference" template is getting too interesting? Or do we prefer to expose as much configuration as possible as a feature reference?

Yes that's my main concern as far as I know some users are actually using that reference template as base and then applying their own kustomizations to it so I would consider those templates we publish as assets to be "best practice" examples and therefore we should avoid using them to configure random values for testing purposes as it may not be clear to a user what is there because it's needed for proper configuration vs. what was added solely for testing purposes

>  We can always simplify these configurations for the reference templates and triage them to the test/ci templates only

That's seems reasonable

Reply:
How do we feel about doing that independent of this PR (cleaning up the reference AKS template so it only has the minimum, required configuration)?

Reply:
That's fine (I see it's already merged) but let's add that to the milestone to make sure we do it before the template gets published as part of the release

Reply:
I think this is what we want: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2913